### PR TITLE
Fix lint/test errors caused by keras 2.6.0

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

Fix lint/test errors caused by keras 2.6.0.

## How is this patch tested?

Existing tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
Filed an issue in the keras repository: https://github.com/keras-team/keras/issues/15147

**Comment by harupy** [comment]
It seems that the early stopping behavior changed:

In keras 2.4.3:

https://github.com/mlflow/mlflow/pull/4679/checks?check_run_id=3345729760#step:11:40

```
{'loss': 1.1561641693115234, 'acc': 0.3199999928474426, 'stopped_epoch': 1.0, 'restored_epoch': 0.0}
```


In keras 2.6.0:


https://github.com/mlflow/mlflow/pull/4679/checks?check_run_id=3345729802#step:11:40

```
{'loss': 1.1771961450576782, 'acc': 0.36000001430511475, 'stopped_epoch': 0.0}
```

File: tests/keras_autolog/test_keras_autolog.py [hunk 1]
```diff
@@ -261,15 +272,24 @@ def test_keras_autolog_log_models_configuration(
     assert ("model" in artifacts) == log_models
 
 
+# In keras 2.6.0, early stopping doesn't work correctly when `patience` is set to 0:
+# https://github.com/keras-team/keras/pull/14750
+patience_values = [1, 5] if keras_version == "2.6.0" else [0, 1, 5]
```
Comment (harupy) [inline tests/keras_autolog/test_keras_autolog.py:277]: Workaround for https://github.com/mlflow/mlflow/pull/4679#issuecomment-899961968

**Comment by harupy** [comment]
@dbczumar

Brief summary of the early stopping issue in keras 2.6.0:

- When `patience` is 0, training early stops at the first epoch, but it actually should not because you can't tell whether the target metric has improved or not at the first epoch.
- https://github.com/keras-team/keras/pull/14750 fixed this issue, but keras 2.6.0 doesn't contain this patch.

## How does this issue affect MLflow?

https://github.com/mlflow/mlflow/blob/92b7ac7b0138588487aeb380d9a210aa70202a7a/mlflow/keras.py#L744

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !
